pkg/cloud/config.go

```diff
@@ -2,297 +2,126 @@ package cloud
-	if c.Token == "" && c.Server.URL == "" && c.Api.URL == "" {
-		return xerrors.New("no config to save, required fields are token, server url, and api url")
-	}
-
-	if err := c.initFirstLogin(); err != nil {
-		return err
-	}
-
-	if err := keyring.Set(ServiceName, TokenKey, c.Token); err != nil {
-		return err
-	}
-
-	configPath := getConfigPath()
-	if err := os.MkdirAll(filepath.Dir(configPath), 0o700); err != nil {
-		return err
-	}
-
-	configYaml, err := yaml.Marshal(c)
-	if err != nil {
-		return err
-	}
-
-	yamlWithFrontmatter := append([]byte("---\n"), configYaml...)
-	return os.WriteFile(configPath, yamlWithFrontmatter, 0o600)
-}
+func GetConfigs(ctx context.Context, opts *flag.Options, accessToken string) error {
+	logger := log.WithPrefix(log.PrefixCloud)
+	client := xhttp.ClientWithContext(ctx)
 
-func Clear() error {
-	if err := keyring.Delete(ServiceName, TokenKey); err != nil {
-		if !errors.Is(err, keyring.ErrNotFound) {
-			return err
+	if opts.CloudOptions.SecretConfig && opts.Scanners.Enabled(types.SecretScanner) {
+		if opts.SecretOptions.SecretConfigPath != "" {
+			logger.Warn("Secret config path already set", log.String("configPath", opts.SecretOptions.SecretConfigPath))
+			return nil
 		}
-	}
 
-	configPath := getConfigPath()
-	if err := os.Remove(configPath); err != nil {
-		if !errors.Is(err, os.ErrNotExist) {
-			return err
+		configPath, err := getConfigFromTrivyCloud(ctx, client, opts, accessToken, ConfigTypeSecret)
+		if err != nil {
+			return xerrors.Errorf("failed to get secret config: %w", err)
 		}
-	}
-
-	return nil
-}
-
-// initFirstLogin initializes the default scanning settings to turn them on
-// after this, the user can configure in the config using the config set/unset commands
-func (c *Config) initFirstLogin() error {
-	if c.Token == "" {
-		// this isn't a login save, without a token it can't login
-		return nil
-	}
-
-	var firstLogin bool
-	_, err := keyring.Get(ServiceName, TokenKey)
-	if err != nil {
-		if !errors.Is(err, keyring.ErrNotFound) {
-			return err
+		if configPath != "" {
+			opts.SecretOptions.SecretConfigPath = configPath
 		}
-		firstLogin = true
-	}
-
-	if firstLogin {
-		// if first login, turn on all scanning options
-		c.Server.Scanning.Enabled = true
-		c.Server.Scanning.UploadResults = true
-		c.Server.Scanning.MisconfigConfig = true
-		c.Server.Scanning.SecretConfig = true
 	}
 	return nil
 }
 
-// Load loads the Trivy Cloud config from the config file and the keychain
-// If the config file does not exist the default config is returned
-func Load() (*Config, error) {
+// getConfigFromTrivyCloud downloads a config from Trivy Cloud and saves it to a file
+// it returns the path to the config file if it was downloaded successfully, otherwise it returns an error
+func getConfigFromTrivyCloud(ctx context.Context, client *http.Client, opts *flag.Options, accessToken string, configType ConfigType) (string, error) {
 	logger := log.WithPrefix(log.PrefixCloud)
```

Comment:
You can create a logger with an attribute so that you don't have to pass it on later:
```suggestion
	logger := log.WithPrefix(log.PrefixCloud).With(log.String("configType", string(configType)))
```